src/home_robot_hw/home_robot_hw/stretch_client/modules/abstract.py

```diff
@@ -18,8 +20,30 @@ def wrapper(self, *args, **kwargs):
 
 
 class AbstractControlModule(abc.ABC):
+    """
+    Abstract control module that implements the following functionalities
+    - Enabling / disabling the module to control access of methods
+    - Concurrency management
+    """
+
     _is_enabled: bool = False
 
+    def __init__(self):
+        self._wait_threads = []
+
+    def _register_wait(self, func):
+        thr = threading.Thread(target=func)
+        self._wait_threads.append()
+        thr.start()
+
+    def is_busy(self):
```

Comment:
might be worth it to add a comment here since i'm not fully sure what is expected to be happening?